### PR TITLE
Add jupyter image folder

### PR DESCRIPTION
**Pull request by alexgleith**
This is a child Docker image, based on the datacube-core base image.

When this is in the repo, we can set up an automatic build, so we've got a standard jupyter build.

A couple of questions, what should the name be? I'm thinking `opendatacube/jupyter` to keep it simple.

### CODE REVIEW

File: jupyter/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    NB_UID=1000 \
+    NB_GID=100
+ENV HOME=/notebooks
+RUN mkdir $HOME
+
+ADD https://raw.githubusercontent.com/jupyter/docker-stacks/master/base-notebook/fix-permissions /usr/local/bin/fix-permissions
+RUN chmod +x /usr/local/bin/fix-permissions 
+# Create user with UID=1000 and in the 'users' group
+# and make sure these dirs are writable by the `users` group.
+RUN useradd -m -s /bin/bash -N -u $NB_UID $NB_USER && \
```
Comment (Kirill888) [inline jupyter/Dockerfile]: If this a different user than one used by datacube image then you need to re-do datacube config files for this user also.

File: jupyter/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
+FROM opendatacube/datacube-core
+
+# Get dependencies for Jupyter
+RUN apt-get update && apt-get install -y \
```
Comment (Kirill888) [inline jupyter/Dockerfile]: I don't think this is needed, you installing `jupyter` from pip, which overrides these things anyway

File: jupyter/Dockerfile (hunk 1 above)
Comment (alexgleith) [reply jupyter/Dockerfile]: Good point, I'll test this properly now. It should still be able to write a config file (which is what the core image does, dynamically). Jut need to make sure the default location is sensible perhaps.

File: jupyter/Dockerfile (hunk 2 above)
Comment (alexgleith) [reply jupyter/Dockerfile]: True, I've removed this.

File: jupyter/Dockerfile (hunk 1 above)
Comment (alexgleith) [reply jupyter/Dockerfile]: And I've added a better way to run it, which actually spins up a system with Docker Compose. Seems to write the .conf file happily. Not sure how this should be managed in a potential multi-user environment, but it's good enough for now, I think.

**Pull request merged**